pkg/karmadactl/cmdinit/kubernetes/deployments.go

```diff
@@ -112,6 +109,11 @@ func (i *CommandInitOption) karmadaAPIServerContainerCommand() []string {
 		fmt.Sprintf("--tls-private-key-file=%s/%s.key", karmadaCertsVolumeMountPath, options.ApiserverCertAndKeyName),
 		"--tls-min-version=VersionTLS13",
 	}
+	if i.isExternalEtcdTLS() || !i.isExternalEtcdProvided() {
```

Comment:
https://github.com/karmada-io/karmada/blob/30627140da43065446ac8114a590b8ec8cb56499/pkg/karmadactl/addons/search/manifests.go#L58-L61

Does it need to be modified here?

Reply:
good catch! let's leave it to the next PR and enabled addons test for karmadactl (i have not see any test for addons with karmadactl),  WDYT?

Reply:
sure, I want to avoid manual testing, so I will also enable addons testing while handling karmada search in this PR.

Reply:
You've added the addons test to the `cli`, but the manifest for search hasn't been changed yet, is the cli failing because of this